Zend/Optimizer/optimize_func_calls.c

```diff
@@ -252,6 +253,13 @@ void zend_optimize_func_calls(zend_op_array *op_array, zend_optimizer_ctx *ctx)
 				if (call_stack[call - 1].func_arg_num != (uint32_t)-1
 						&& has_known_send_mode(&call_stack[call - 1], call_stack[call - 1].func_arg_num)) {
 					if (ARG_SHOULD_BE_SENT_BY_REF(call_stack[call - 1].func, call_stack[call - 1].func_arg_num)) {
+						/* There's no TMP specialization for FETCH_OBJ_W/FETCH_DIM_W. Avoid
+						 * converting it and error at runtime in the FUNC_ARG variant. */
+						if ((opline->opcode == ZEND_FETCH_OBJ_FUNC_ARG || opline->opcode == ZEND_FETCH_DIM_FUNC_ARG) && opline->op1_type == IS_TMP_VAR) {
+							/* Don't remove the associated CHECK_FUNC_ARG opcode. */
+							call_stack[call - 1].check_func_arg_opline = NULL;
+							break;
+						}
```

Comment:
It's possible to have few ZEND_FETCH_DIM_FUNC_ARG in a row (byRef(C[0][0][0]). You are going to keep the first but cheche the second. Right?

Reply:
Yes. I think this shouldn't matter (in terms of the VM, at least), as the first one will always throw.

Reply:
> Yes. I think this shouldn't matter (in terms of the VM, at least), as the first one will always throw.

It's better to check if ``call_stack[call - 1].check_func_arg_opline`` was reset before and prevent invalid optimizations.



Yes. I think this shouldn't matter (in terms of the VM, at least), as the first one will always throw.

Reply:
Alright, that makes sense.